### PR TITLE
ENH: add user and assistant name to chat

### PR DESCRIPTION
**Pull request by RayJi01**

### CODE REVIEW

File: xinference/model/llm/core.py [hunk 1]
```diff
@@ -220,15 +220,17 @@ def _to_prompt(
         self,
         prompt: str,
         system_prompt: str,
+        username,
```
Comment (UranusSeven) [inline xinference/model/llm/core.py:223]: Type hint.

File: xinference/model/llm/core.py [hunk 2]
```diff
@@ -182,7 +182,7 @@ def generator_wrapper(
             for _completion_chunk in self._llm(prompt=_prompt, **_generate_config):
                 yield _completion_chunk
 
-        logger.debug(
+        logger.error(
```
Comment (UranusSeven) [inline xinference/model/llm/core.py:185]: Restore code for debugging.

File: xinference/model/llm/core.py [hunk 3]
```diff
@@ -140,7 +140,7 @@ def _sanitize_model_config(
         else:
             context_length = 2048
 
-        llamacpp_model_config.setdefault("n_gpu_layers", self._gpu_layers)
+        # llamacpp_model_config.setdefault("n_gpu_layers", self._gpu_layers)
```
Comment (UranusSeven) [inline xinference/model/llm/core.py:143]: Restore code for debugging.

File: xinference/model/llm/core.py (hunk 2 above)
Comment (RayJi01) [reply xinference/model/llm/core.py]: resolved in the latest version

File: xinference/model/llm/core.py (hunk 3 above)
Comment (RayJi01) [reply xinference/model/llm/core.py]: resolved in the latest version

File: xinference/model/llm/core.py (hunk 1 above)
Comment (RayJi01) [reply xinference/model/llm/core.py]: Resolved in the latest version

**Review by UranusSeven — APPROVED** [review]

**Comment by UranusSeven** [comment]
There's not a OpenAI compatible way to pass the user and assistant name. 

In this case, we need to use the low level API `generate` and piece together a prompt based on the chat history.

**Pull request closed without merging**